frontend/src/pages/ArtifactList.tsx

```diff
@@ -157,29 +157,25 @@ class ArtifactList extends Page<{}, ArtifactListState> {
     <ArtifactLink artifactUri={value} />
   );
 
+  private async getArtifacts(): Promise<Artifact[]> {
+    const response = await this.api.metadataStoreService.getArtifacts(new GetArtifactsRequest());
+
+    if (!response) {
+      this.showPageError('Unable to retrieve Artifacts.');
+      return [];
+    }
+
+    return response!.getArtifactsList() || [];
+  }
+
   /**
    * Temporary solution to apply sorting, filtering, and pagination to the
    * local list of artifacts until server-side handling is available
    * TODO: Replace once https://github.com/kubeflow/metadata/issues/73 is done.
    * @param request
    */
-  private async getRowsFromArtifacts(request: ListRequest, artifacts: Artifact[]): Promise<void> {
-    const artifactTypesMap = new Map<number, ArtifactType>();
-    // TODO: Consider making an Api method for returning and caching types
-    const {
-      response: res,
-      error: err,
-    } = await Apis.getMetadataServicePromiseClient().getArtifactTypes(
-      new GetArtifactTypesRequest(),
-    );
-    if (err) {
-      this.showPageError(serviceErrorToString(err));
-      return;
-    }
-
-    ((res && res.getArtifactTypesList()) || []).forEach(artifactType => {
-      artifactTypesMap.set(artifactType.getId()!, artifactType);
-    });
+  private async getRowsFromArtifacts(request: ListRequest): Promise<void> {
+    const artifacts = [...this.state.artifacts];
```

Comment:
same problem as commented below, artifacts were just set with `setState`, there's no guarantee the artifacts in state is already up-to-date now.
Recommend passing from arguments.

Reply:
Removed the `setState` call